src/main/java/seedu/address/storage/JsonAdaptedTutorial.java

```diff
@@ -83,6 +101,16 @@ public Tutorial toModelType() throws IllegalValueException {
         }
         final Time modelTime = new Time(time);
 
-        return new Tutorial(modelTutorialName, modelVenue, modelDay, modelTime);
+        final AssessmentResultsList modelAssessmentResultsList = new AssessmentResultsList(modelTutorialName);
+        for (JsonAdaptedAssessmentResults assessmentResults : assessmentResultsList) {
+            AssessmentName assessmentName = assessmentResults.getAssessmentName();
+            if (!assessmentList.containsByName(assessmentName)) {
+                throw new IllegalValueException(MISSING_ASSESSMENT);
+            }
+            modelAssessmentResultsList.add(assessmentResults.toModelType(
+                    assessmentList.getByName(assessmentName).getFullMark()));
+        }
+
```

Comment:
This feels a bit complex, will it be better to break it down?

Reply:
i'll add a comment explaining it!